packages/coinstac-ui/app/render/components/results/displays/images.jsx

```diff
@@ -37,48 +71,199 @@ class Images extends Component {
+              </h4>
+            );
             if (typeof w === 'string') {
-              output.push(<img alt={`${w}-img`} src={`data:image/png;base64, ${w}`} />);
+              subitem.push(
+                <img alt={`${w}-img`} src={`data:image/png;base64, ${w}`}
+                  style={styles.image} />
+              );
+            }
+            if (h + 1 <= subItemsLength/2 && subcol1.length < subItemsLength/2) {
+              subcol1.push(
+                <div className={'item'} key={`image-item-${key}-${k}-${l}`} style={styles.imgItem}>{subitem}</div>
+              );
+            }
+            if (h + 1 > subItemsLength/2  && subcol2.length < subItemsLength/2) {
+              subcol2.push(
+                <div className={'item'} key={`image-item-${key}-${k}-${l}`} style={styles.imgItem}>{subitem}</div>
+              );
             }
           });
+        };
+        if (key.includes('global')) {
+          if (i + 1 <= itemsLength/2) {
+            col1.push(
+              <div className={'item'} key={`page-${key}-${k}`} style={styles.imgItem}>{item}</div>
+            );
+          }
+          if (i + 1 > itemsLength/2) {
+            col2.push(
+              <div className={'item'} key={`page-${key}-${k}`} style={styles.imgItem}>{item}</div>
+            );
+          }
+        }else{
+          page.push(
+            <div className={`page-${k}`} ref={`page-${k}`} key={`page-${k}`} style={styles.subpage}>
+              {item}
+              <div style={styles.column}>{subcol1}</div>
+              <div style={styles.column}>{subcol2}</div>
+            </div>
+          );
+          subcol1 = [];
+          subcol2 = [];
         }
       });
+      page.push(
+        <div style={styles.column}>{col1}</div>
+      );
+      page.push(
+        <div style={styles.column}>{col2}</div>
+      );
+      if (key.includes('global')) {
+        output.push(
+          <div key={`page-${key}`} className={`page-${key}`} ref={'global_page'} style={styles.page}>{page}</div>
+        );
+      }else{
+        output.push(
+          <div key={`page-${key}`} className={'local_stats'} ref={`local_page_${key}`} style={styles.page}>{page}</div>
+        );
+      }
     });
     return output;
   }
 
+  renderCanvas = () => {
+    const { plotData } = this.props;
+    let global_canvas = ReactDOM.findDOMNode(this.refs.global_canvas);
+    let global_results = ReactDOM.findDOMNode(this.refs.global_page);
+    RasterizeHTML.drawHTML(global_results.innerHTML, global_canvas);
+    Object.entries(plotData.local_stats).forEach(([key, value]) => {
+        let canvas = key + '_canvas';
+        let page = 'page-' + key;
+        let key_canvas = ReactDOM.findDOMNode(this.refs[canvas]);
+        let key_results = ReactDOM.findDOMNode(this.refs[page]);
+        RasterizeHTML.drawHTML(key_results.innerHTML, key_canvas);
+    });
+  }
+
+  componentDidMount = () => {
+    setTimeout(this.renderCanvas, 1000);
+  }
+
+  savePDF = () => {
```

Comment:
Wow, surprisingly little code to get a PDF! That's neat.

Reply:
Thanks!